data/filter/FieldFilter.js

```diff
@@ -87,79 +87,72 @@ export class FieldFilter extends Filter {
-                    return !isNil(v) && v >= value;
-                };
+                opTestFn = v => !isNil(v) && v >= value;
+                break;
             case '<':
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    const v = getVal(r);
-                    return !isNil(v) && v < value;
-                };
+                opTestFn = v => !isNil(v) && v < value;
+                break;
             case '<=':
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    const v = getVal(r);
-                    return !isNil(v) && v <= value;
-                };
+                opTestFn = v => !isNil(v) && v <= value;
+                break;
             case 'like':
                 regExps = value.map(v => new RegExp(escapeRegExp(v), 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    return regExps.some(re => re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.some(re => re.test(v));
+                break;
             case 'not like':
                 regExps = value.map(v => new RegExp(escapeRegExp(v), 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    regExps.every(re => !re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.every(re => !re.test(v));
+                break;
             case 'begins':
                 regExps = value.map(v => new RegExp('^' + escapeRegExp(v), 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    regExps.some(re => re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.some(re => re.test(v));
+                break;
             case 'ends':
                 regExps = value.map(v => new RegExp(escapeRegExp(v) + '$', 'i'));
-                return r => {
-                    if (doNotFilter(r)) return true;
-                    regExps.some(re => re.test(getVal(r)));
-                };
+                opTestFn = v => regExps.some(re => re.test(v));
+                break;
             default:
                 throw XH.exception(`Unknown operator: ${op}`);
         }
+
+        if (!store) return r => opTestFn(r[field]);
+
+        // Pass/fail depends on the filter value mode of the store
+        // Uncommitted records always pass the filter
+        switch (store.filterValueMode) {
+            case FilterValueMode.COMMITTED:
+                // Pass only if the committed value passes
+                return r => r.isAdd || opTestFn(r.getCommitted(field));
+            case FilterValueMode.CURRENT:
+                // Pass only if the current value passes
+                return r => r.isAdd || opTestFn(r.get(field));
+            case FilterValueMode.ANY:
+                // Pass if either the current or committed value passes
```

Comment:
running the test twice here in the any case, even if -- the most likely case --- there is no edit and the values are the same.  We may want to optimize that.